build-info-extractor/src/main/java/org/jfrog/build/extractor/issuesCollection/IssuesCollector.java

```diff
@@ -0,0 +1,147 @@
+        IssuesCollectionConfig parsedConfig = parseConfig(config);
+        String previousVcsRevision = getPreviousVcsRevision(client, buildName);
+        Set<Issue> affectedIssues = doCollect(execDir, logger, parsedConfig, previousVcsRevision);
+        return buildIssuesObject(parsedConfig, affectedIssues);
+    }
+
+    IssuesCollectionConfig parseConfig(String config) throws IOException {
+        // When mapping the config from String to IssuesCollectionConfig one backslash is being removed, multiplying the backslashes solves this.
+        config = config.replace("\\", "\\\\");
+        ObjectMapper mapper = new ObjectMapper(new JsonFactory());
+        IssuesCollectionConfig parsedConfig = mapper.readValue(config, IssuesCollectionConfig.class);
+        parsedConfig.validateConfig();
+        return parsedConfig;
+    }
+
+    /**
+     * Gets the previous vcs revision from the LATEST build published to Artifactory.
+     * */
+    private String getPreviousVcsRevision(ArtifactoryBuildInfoClient client, String buildName) throws IOException {
+        // Get LATEST build info from Artifactory
+        Build previousBuildInfo = client.getBuildInfo(buildName, LATEST);
+        if (previousBuildInfo == null) {
+            return "";
+        }
+        if (StringUtils.isNotEmpty(previousBuildInfo.getVcsRevision())) {
+            return previousBuildInfo.getVcsRevision();
+        }
+        // If revision is not listed explicitly, get revision from the first not empty Vcs of the Vcs list.
+        List<Vcs> vcsList = previousBuildInfo.getVcs();
+        if (vcsList != null && vcsList.size() > 0) {
+            for (Vcs curVcs : previousBuildInfo.getVcs()) {
+                if (StringUtils.isNotEmpty(curVcs.getRevision())) {
+                    return curVcs.getRevision();
+                }
+            }
+        }
+        return "";
+    }
+
+    /**
+     * Collects affected issues from git log
+     */
+    private Set<Issue> doCollect(File execDir, Log logger, IssuesCollectionConfig issuesConfig, String previousVcsRevision) throws InterruptedException, IOException {
+        verifyGitExists(execDir, logger);
+        String gitLog = getGitLog(execDir, logger, previousVcsRevision);
+
+        int keyIndex = issuesConfig.getIssues().getKeyGroupIndex();
+        int summaryIndex = issuesConfig.getIssues().getSummaryGroupIndex();
+        Set<Issue> affectedIssues = new HashSet<>();
+
+        Pattern pattern = Pattern.compile(issuesConfig.getIssues().getRegexp());
+        String[] lines = gitLog.split("\\R");
+        for (String line : lines) {
+            Matcher matcher = pattern.matcher(line);
+            if (matcher.find()) {
+                Issue foundIssue = getMatchingIssue(keyIndex, summaryIndex, matcher, issuesConfig);
+                affectedIssues.add(foundIssue);
+                logger.debug("Found issue: " + foundIssue.getKey());
```

Comment:
I suggest changing this from *debug* to *info* and change the message to:
"Added issue XXX to the build-info."